src/core/dense_set.cc

```diff
@@ -173,6 +172,30 @@ bool DenseSet::Equal(DensePtr dptr, const void* ptr, uint32_t cookie) const {
   return ObjEqual(dptr.GetObject(), ptr, cookie);
 }
 
+bool DenseSet::IsEmpty(uint32_t bid) const {
```

Comment:
I would change the name of the function. like `IsBucketBusy` or something. 

Also, you may want to strengthen conditions about the displacement for each of the entries.